### ISSUE TITLE
Kraken -> TypeError: Cannot read property 'fee' of undefined

### ISSUE THREAD

**Issue by paulomalvar** [5an1ty/BitBot#21]
After configuring everything as instructed I run "node app.js -b" and get the following error type:

TypeError: Cannot read property 'fee' of undefined

Not sure what to do in order to get this fix. Any help will be appreciated.

**Comment by tabenius** [comment]
I got that too. Then I modified BitBot to check if that is the case then set fee=0. However, then the next error is 
<  var highestBid = _.first(result.orderBook.bids).currencyPrice;
<                                            ^
< TypeError: Cannot read property 'bids' of undefined

without bids from the orderbook the application would be pointless 👎 

Is this due to some kraken API change or how can BitBot fail so miserably all of a sudden?

**Comment by 5an1ty** [comment]
I just pulled a clean install of BitBot and cannot reproduce this issue.

Can you provide any more info on this? What Configuration are you using? Are you sure your api key / secret combination is still valid?

**Comment by 5an1ty** [comment]
If I try with an unknown assetpair there should be a clean error like this:

[05-07-2016 18:43:34] - ERROR: getOrderBook: Kraken API returned Unknown asset pair error, exiting!

So I'm really wondering what your or @paulomalvar configs looked like.

**Comment by tabenius** [comment]
Now I made a clean clone and in config.js set
 14   exchange: 'kraken',
 15   // Options: (bitstamp, kraken, btce)
 16   currencyPair: {pair: 'XBTUSD', asset: 'XBT', currency: 'USD'},

and get 
[...]BitBot.clean/exchanges/kraken.js:198
        }).fee);
          ^
TypeError: Cannot read property 'fee' of undefined

**Comment by 5an1ty** [comment]
That pair is indeed incorrect it should be "XXBTZUSD":

`currencyPair: {pair: 'XXBTZUSD', asset: 'XXBT', currency: 'ZUSD'}`

I'm not sure why I don't return a clean error in this case as it should. I probably expect a response from the API that I'm clearly not getting when using XBTUSD.

Currencypairs for kraken can be looked up here:
https://api.kraken.com/0/public/AssetPairs

**Comment by tabenius** [comment]
Yes, that is very odd. However, even if you don't get a clean error it would still be good to check if
          var cp = _.find(data.result.fees, function (value, key) { return key === pair; });
is defined before trying to find it's fee.

**Comment by 5an1ty** [comment]
I agree, I'll look into handling that better!

**Comment by larchanka** [comment]
It means you've entered incorrect currencyPair
